crates/forge/tests/e2e/docs_snippets_validation.rs

```diff
@@ -1,43 +1,96 @@
-use regex::Regex;
+use shared::test_utils::output_assert::assert_stdout_contains;
+
+use super::common::runner::{
+    setup_hello_workspace, setup_package, setup_package_from_docs_listings, test_runner,
+};
+
+fn is_package_from_docs_listings(
+    package: &str,
+    listings_to_packages_mapping: &HashMap<String, Vec<String>>,
+) -> bool {
+    for packages in listings_to_packages_mapping.values() {
+        if packages.contains(&package.to_string()) {
+            return true;
+        }
+    }
+    false
+}
+
 #[test]
 fn test_docs_snippets() {
+    let listings_to_packages_mapping = create_listings_to_packages_mapping();
+
     let root_dir = get_parent_dir(2);
     let docs_dir = root_dir.join("docs/src");
 
-    let re = Regex::new(r"(?ms)```shell\n\$ (snforge .+?)\n```").expect("Invalid regex pattern");
+    let snippet_type = SnippetType::Forge;
 
-    let snippets = extract_snippets_from_directory(&docs_dir, &re)
-        .expect("Failed to extract snforge command snippets");
+    let snippets = extract_snippets_from_directory(&docs_dir, &snippet_type)
+        .expect("Failed to extract command snippets");
 
     // TODO(#2684)
     let skipped_args = [
-        // for some reason `try_parse_from` fails on `--version` flag
+        // For some reason `try_parse_from` fails on `--version` flag, it returns Err but produces the expected output
         vec!["snforge", "--version"],
     ];
 
     for snippet in &snippets {
         let args = snippet.to_command_args();
-        let args: Vec<&str> = args.iter().map(String::as_str).collect();
+        let mut args: Vec<&str> = args.iter().map(String::as_str).collect();
 
         if skipped_args.contains(&args) {
-            print_skipped_snippet_message(snippet, "snforge");
+            print_skipped_snippet_message(snippet);
             continue;
         }
 
-        let parse_result = Cli::try_parse_from(args);
+        let parse_result = Cli::try_parse_from(args.clone());
         let err_message = if let Err(err) = &parse_result {
             err.to_string()
         } else {
             String::new()
         };
-        assert_valid_snippet(parse_result.is_ok(), snippet, "snforge", &err_message);
+
+        assert_valid_snippet(parse_result.is_ok(), snippet, &err_message);
+
+        // Remove "snforge" from the args
+        args.remove(0);
+
+        // Remove "test" from the args
+        args.retain(|element| element != &"test");
+
+        if let Some(snippet_output) = &snippet.output {
+            let package_name = snippet
+                .capture_package_from_output()
+                .expect("Failed to capture package from command output");
+
+            let temp =
+                if is_package_from_docs_listings(&package_name, &listings_to_packages_mapping) {
+                    setup_package_from_docs_listings(&package_name, &listings_to_packages_mapping)
+                } else {
+                    resolve_package_name(&package_name)
+                };
```

Comment:
note: Currently we use `hello_workpaces` package (from `crates/forge/tests/data` in one example in docs. In future we can simplify this logic, once we use only packages from docs listings.